bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -300,28 +404,44 @@ private int handleUpdates(List<Update> updates) {
-                    && update.callbackQuery().message().text() != null) {
-                String[] callbackData = update.callbackQuery().data().split(" ", 2);
+                if (lastMessageURL != null || lastMessageText != null) {
+                    lastMessageDate = message.date();
+                    lastMessageFirstName = message.from().firstName();
+                    lastMessageLastName = message.from().lastName();
+                    lastMessageUsername = message.from().username();
+                }
+            } else if (callbackQuery != null && callbackQuery.message() != null
+                    && callbackQuery.message().text() != null) {
+                String[] callbackData = callbackQuery.data().split(" ", 2);
 
                 if (callbackData.length == 2) {
                     replyId = callbackData[0];
                     lastMessageText = callbackData[1];
-                    lastMessageDate = update.callbackQuery().message().date();
-                    lastMessageFirstName = update.callbackQuery().from().firstName();
-                    lastMessageLastName = update.callbackQuery().from().lastName();
-                    lastMessageUsername = update.callbackQuery().from().username();
-                    chatId = update.callbackQuery().message().chat().id();
-                    replyIdToCallbackId.put(new ReplyKey(chatId, replyId), update.callbackQuery().id());
-                    logger.debug("Received callbackId {} for chatId {} and replyId {}", update.callbackQuery().id(),
-                            chatId, replyId);
+                    lastMessageDate = callbackQuery.message().date();
+                    lastMessageFirstName = callbackQuery.from().firstName();
+                    lastMessageLastName = callbackQuery.from().lastName();
+                    lastMessageUsername = callbackQuery.from().username();
+                    chatId = callbackQuery.message().chat().id();
+                    replyIdToCallbackId.put(new ReplyKey(chatId, replyId), callbackQuery.id());
+
+                    // build and publish callbackEvent trigger channel payload
+                    JsonObject callbackRaw = json.parse(gson.toJson(callbackQuery)).getAsJsonObject();
+                    JsonObject callbackPayload = new JsonObject();
+                    callbackPayload.addProperty("message_id", callbackQuery.message().messageId());
+                    callbackPayload.addProperty("from", String.join(" ",
+                            new String[] { callbackQuery.from().firstName(), callbackQuery.from().lastName() }));
+                    callbackPayload.addProperty("chat_id", callbackQuery.message().chat().id());
+                    callbackPayload.addProperty("callback_id", callbackQuery.id());
+                    callbackPayload.addProperty("reply_id", callbackQuery.data().split(" ", 2)[0]);
+                    callbackPayload.addProperty("text", callbackQuery.data().split(" ", 2)[1]);
```

Comment:
Please cache `callbackQuery.data().split(" ", 2)` to a local variable so it isn't called twice.

Reply:
Resolved in [7fe68eb](https://github.com/openhab/openhab-addons/pull/9251/commits/7fe68eb54b09bc6a88bc1fd434b335a63ac1ae82)